content/ADR-125-realm-modifiers-for-worlds-in-renderer.md

```diff
@@ -0,0 +1,196 @@
+
+Discoverability of content will be outside-in, as content creators will be able to share easy to remember URLs to enter the World directly, or inside-out using a join command. Anybody will be able to join a World.
+
+
+## Solution Space Exploration
+
+To be able to modify the features necessary in each world, we are developing a Realms Plugin which contains Realms Modifiers. Each of this modifiers will have the responsibility to alter DCL functionality by checking if we are entering a user-generated realm or a DAO realm. 
+
+## Specification
+
+### Realms Plugin
+
+The realms plugin is the entry point for the realms modifier. Its is created under the `explorer-realms_modifier_plugin` flag. It's the bridge which receives the 'AboutResponse' sent by kernel and distributes it among all implemented modifiers. By comparing the just accessed realms with the list of available catalysts, it determines if a realm is part of the DAO or a user created world. This variable is necessary for modifiers to execute certain functionaly. On the near future, we could add this bool to the 'AboutResponse' section, so this responsability relies complete on the data and not on a comparison.
+
+### Realms Modifiers Interface
+
+Describes the behaviour necessesary for realms modifiers. Its current implementation requires to specify if we are on a DAO realm; since some behaviour still relies on this property
+
+```
+namespace DCLPlugins.RealmsPlugin
+{
+    public interface IRealmsModifier : IDisposable
+    {
+        void OnEnteredRealm(bool isCatalyst, AboutResponse realmConfiguration);
+    }
+}
+```
+
+ The current modifiers currently implemented area:
+
+- **Realms Blocker Modifier**
+
+Controls how the green blockers should be shown on each realm. If a realm is a Catalyst, or it has a value on Configurations.CityLoaderContentServer; then the green blockers will be shown.
+
+- **Realms Minimap Modifier**
+
+Controls the functionality associated to the minimap when entering a Realm. It determines if the minimap should be visible using the realmConfiguration.Configurations.Minimap.Enabled value. Also, it determines if the Jump To Home Button should be present. This button will appear on any given user world, and it will take the user back to the home location in the most populated catalyst. 
+When needed, this modifier will implement the change of the minimap image; as a user generated world will have the possibility to show its own minimap png.
+	
+
+### Data Structure 
+
+`AboutResponse` contains the info of the realm which has just been accessed. This class is autogenerated according to protobuff specifications. Each modifier gets access to the AboutResponse, and gets the info necessary to execute its behaviour. It has the following structure: 
```

Comment:
It's not clear to me why renderer needs this `AboutResposne`.

Reply:
The `AboutResponse.configurations` contains the state necessary for the current modifiers. I will add an example to make it more clear; because I see that the one below doesn't have the `minimap` variable or `cityLoaderContentServer` variable; both are used in the current modifiers.

On the other hand, we discussed with @menduz sending the `AboutResponse` to replace the use of the `RealmsInfo`, since they are containing redundant info.  The redudant info that currently being sent is:

1) `content.publicURL` and `lambdas.publicURL`  are used to set the Catalyst info in renderer;
2) `configurations.realmName` is used for ExploreV2Menu to show the name of the current realm.

Right now they are co-existing, but whenever we want, we can remove the `UpdateRealmsInfo` call from Kernel.
![image](https://user-images.githubusercontent.com/1999557/203544916-bd4abc57-b7dc-4a69-b8c2-57b03672c497.png)